napari/components/layerlist.py

```diff
@@ -93,6 +96,9 @@ def _create_contexts(self):
 
         self._ctx = create_context(self)
         if self._ctx is not None:  # happens during Viewer type creation
+            self._ctx['selected_empty_shapes_layer'] = (
```

Comment:
Hmmm so even though the `ContextMapping` can evaluate the function without any issue, we still have to declare this context key separately from the other layerlist ones? Why can't it just be listed alongside the other `LayerListSelectionContextKeys`?

Reply:
Because this function depends on Qt. And `LayerListSelectionContextKeys` lives in non-Qt namespace.

Reply:
See https://github.com/napari/napari/pull/6965#discussion_r1629152571

Reply:
I think if we define this context key up in `_layerlist_context`, we don't need to define it again here. When I commented it out locally everything seemed to still work as usual

Reply:
I agree